base/string.jl

```diff
@@ -968,8 +968,8 @@ byte_string_classify(s::ByteString) = byte_string_classify(s.data)
     # 1: valid ASCII
     # 2: valid UTF-8
 
-is_valid_ascii(s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) == 1
-is_valid_utf8(s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) != 0
+isvalid(::Type{ASCIIString}, s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) == 1
```

Comment:
It's not an ultimata at all... just the facts.  I'd love to spend a lot more time improving these areas of Julia, but I've also got to keep my client happy... and there are only so many hours in the day.
I've responded as quickly as possible to any recommendations by reviewers, and since I haven't gotten any further feedback, I would assume that there would be no reason not to merge in these changes (I've added a lot more testing than was there previously also, so that can't be an issue).